data/moves.ts

```diff
@@ -15264,13 +15256,13 @@ export const Moves: {[moveid: string]: MoveData} = {
 		pp: 15,
 		priority: 0,
 		flags: {protect: 1, reflectable: 1, mirror: 1, mystery: 1},
-		onTryHit(pokemon) {
-			const bannedAbilities = [
-				'asoneglastrier', 'asonespectrier', 'battlebond', 'comatose', 'disguise', 'multitype', 'powerconstruct', 'rkssystem', 'schooling', 'shieldsdown', 'simple', 'stancechange', 'truant', 'zenmode',
-			];
-			if (bannedAbilities.includes(pokemon.ability)) {
+		onTryHit(target) {
+			if (target.getAbility().neverChanges || target.ability === 'simple') {
 				return false;
 			}
+			if (target.ability === 'truant') {
+				return false; // TODO how to return differently?
```

Comment:
- If the immunity should be checked before accuracy (if relevant), use `onTryImmunity` instead of `onTryHit`.
- If the immunity should be checked here, then return `null`, but also add code to show the target was immune.

Reply:
Yep, onTryImmunity works perfectly for what I want, thanks!